lib/controllers/frontend/spree/checkout_controller_decorator.rb

```diff
@@ -33,11 +33,18 @@ def check_authorization
     # Introduces a registration step whenever the +registration_step+ preference is true.
     def check_registration
       return unless Spree::Auth::Config[:registration_step]
-      return if spree_current_user or current_order.email
+      return if (
+        spree_current_user ||
+        (guest_authenticated? && Spree::Config[:allow_guest_checkout])
```

Comment:
Can this config check be folded into the body of `guest_authenticated?`, or is there a reason you kept them separate?